csi/exporter.py

```diff
@@ -67,12 +83,33 @@ def collect(self):
                 inodes_free_count.add_metric(labels, free)
                 inodes_used_count.add_metric(labels, used)
 
+                # Gathers capacity metrics for each subvol
+                pvcpath = pathlib.Path(os.path.join(pth, PV_TYPE_SUBVOL)).glob("*/*/*")
+                for subvol in pvcpath:
+                    if pathlib.Path(subvol).is_dir():
+                        pvcname = os.path.basename(subvol)
+                        pvclabels = labels + [pvcname]
+                        try:
+                            # Get extended attributes for pvc
+                            size = os.getxattr(subvol, "trusted.gfs.squota.size")
+                            limit = os.getxattr(subvol, "trusted.gfs.squota.limit")
+                            subvol_capacity_bytes.add_metric(pvclabels, limit)
+                            subvol_capacity_used_bytes.add_metric(pvclabels, size)
+                            # Convert bytes type into int to compute the free capacity of the subvol
+                            free = int(limit.decode("utf-8")) - int(size.decode("utf-8"))
```

Comment:
I feel there is a minor issue here. The `trusted.gfs.squota.size` is a value set in 'each' brick/storage pod. When more than one storage volume is given in the mount, this would be returning from one of the brick only. Hence, `os.statvfs()` is  better to call with pvc path to get both total size, and also proper usage.

Reply:
+1 to `os.statvfs`

Reply:
Yes indeed, this is a misunderstanding on my part. I didn't think that `os.statvfs` would get the glusterfs quotas at the pvc directory level.